var/spack/repos/builtin/packages/suite-sparse/package.py

```diff
@@ -65,19 +68,20 @@ def install(self, spec, prefix):
             'CC=%s' % self.compiler.cc,
             'CXX=%s' % self.compiler.cxx,
             'F77=%s' % self.compiler.f77,
-            'CUDA_ROOT     =',
-            'GPU_BLAS_PATH =',
-            'GPU_CONFIG    =',
-            'CUDA_PATH     =',
-            'CUDART_LIB    =',
-            'CUBLAS_LIB    =',
-            'CUDA_INC_PATH =',
-            'NV20          =',
-            'NV30          =',
-            'NV35          =',
-            'NVCC          = echo',
-            'NVCCFLAGS     =',
+            # They "autodetect" CUDA before CUDA_PATH is checked
+            # setting CUDA=no disables them using `which nvcc`,
+            # so that we can use our spack managed cuda path.  If
+            # +cuda we set CUDA_PATH, and the other CUDA specific
+            # libraries are setup by suite-sparse.  See the file
+            # SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk
+            'CUDA=no'
```

Comment:
oh, that's a very non-intuitive logic from developer's side. Please add a short comment above this line, something like

> CUDA=no does NOT disable cuda, it only disables internal search for CUDA_PATH. 
> If in addition the latter is empty, then CUDA is completely disabled.

And then I would combine the two together as
```
'CUDA=no'
'CUDA_PATH=%s' % (spec['cuda'].prefix if '+cuda' in spec 
    else '')
```
to make sure that `CUDA_PATH` is not picked up accidentally from user's environment.
(i did not test the python code, but i hope it's clear what I mean.